config/routes.js

```diff
@@ -1,32 +1,82 @@
-const axios = require('axios');
+const axios = require("axios");
+const bcrypt = require("bcryptjs");
+const db = require("../database/dbConfig");
+const jwt = require("jsonwebtoken");
 
-const { authenticate } = require('../auth/authenticate');
+const { authenticate } = require("../auth/authenticate");
 
 module.exports = server => {
-  server.post('/api/register', register);
-  server.post('/api/login', login);
-  server.get('/api/jokes', authenticate, getJokes);
+  server.post("/api/register", register);
+  server.post("/api/login", login);
+  server.get("/api/jokes", authenticate, getJokes);
 };
 
 function register(req, res) {
   // implement user registration
+  const user = req.body;
+  if (user.username && user.password) {
+    user.password = bcrypt.hashSync(user.password, 5);
+    db("users")
+      .insert(user)
+      .then(([id]) => {
+        db("users")
+          .where({ id: id })
+          .first()
+          .then(user => res.status(200).json(user))
+          .catch(err => res.status(500).json(err));
+      })
+      .catch(err => res.status(500).json(err));
+  } else {
+    res.status(403).json({ message: "required fields were not provided" });
+  }
 }
 
 function login(req, res) {
   // implement user login
+  const { username, password } = req.body;
+
+  db("users")
+    .where({ username: username })
```

Comment:
```suggestion
    .where({ username })
``` unecessary since both key and value are the same name!